### ISSUE TITLE
.Net Core 2.2 Compatabiliity

### ISSUE THREAD

**Issue by bradyclifford** [paraspatidar/NugetDownloader#1]
What is your roadmap for .Net Core 2.2 and future .Net releases?

**Comment by paraspatidar** [comment]
Though Nuget downloader v2.0 is in .netcore 2.1 ,but I am already using it in one of my project which is in .netcore 2.2.

Having said that , converting Nuget downloader itself in .net core 2.2 should not be an issue wrt to code changes, however I shall give it a try and release some 
Netcore 2.2 version as well if there are any related changes.

**Comment by SperoSophia** [comment]
Nuget.PackageManagement is targeting .NET Framework 4.6+ which means that it is not really .NET Core to start with if its dependencies are .NET Framework Libraries

**Comment by paraspatidar** [comment]
Yes , that is correct and nuget' current nuget.package manager is still
with 4.6+  and thus for dotnet core 2.2 it works well, event with
containers and kubernetes and windows deployment

However 4.6+ is the reason it might not work with future.

However there are two ways:
1. Wait for nuget to give its core packages compatible with dot net core.
2. Meanwhile I am working on another project wich is independent of core
packages thus giving ability to make it truely dotnet core compatible.

**Comment by SperoSophia** [comment]
cool, looking forward for the independent package, this might help is a nuget resolver from one of my projects. https://github.com/SperoSophia/Cursive/blob/master/Cursive/PackageManager.cs
Think the nuget package resolver itself has too many dependencies.
the nuget packages themselves are a glorified zip file with an xml with dependency information and some files.

**Comment by SperoSophia** [comment]
Looking forward to the independent nuget resolver, personally i think that the current downloader has too many heavy dependencies.
I took out the nuget resolver from a private repository and shared it on gist, hope that helps. 
https://gist.github.com/SperoSophia/2085340092416da96a5312bfdd744a78

the nuget packages are after all just a glorified zip file with an xml information directory of self reliant dependencies with some files. after figuring this out, i wrote a simple class that downloads the nuget package, extracts its file of the given target framework and does so recursively down until all child dependencies have been resolved, it does not check if these files already exists on GA, but simply assumes that you are running a clean instance. so you might have fun implementing those checks. It worked fine for my needs when i wrote https://Cursive.dev which was a .NET Language (joke language) which needed to be able to resolve nuget dependencies on all platforms. (.NET Core targets)

**Comment by knight1219** [comment]
@Theoistic That gist doesnt work anymore, do you have a working link by chance?

**Comment by knight1219** [comment]
I just submitted a PR for updating the lib to netstandard2.1, so that should be able to hit .net core 3.1 and 2.2(i think 2.2 is compatible but may need to check)

**Comment by Theoistic** [comment]
yeah, sorry changed my username, here is an updated link: https://gist.github.com/Theoistic/2085340092416da96a5312bfdd744a78

**Comment by engrmalikmubashir** [comment]
Sorry to bother you all, but could someone please do a video tutorial on how to use this package (NugetDownloader). I will be very gratefull to you. Thanks in anticipation.
